CorsixTH/Src/main.cpp

```diff
@@ -166,19 +170,29 @@ int CorsixTH_lua_main_no_eval(lua_State *L)
     "end \n"
     "return loadfile(name)";
 
-    // return assert(loadfile"CorsixTH.lua")(...)
+
+
     if(!bGotScriptFile)
     {
+    	// Prepare the stack for calling: assert(LoadFile(CorsixTH.lua)):
         lua_getglobal(L, "assert");
         luaL_loadbuffer(L, sLuaCorsixTHLua, strlen(sLuaCorsixTHLua),
-            "@main.cpp (l_main bootstrap)");
+                        "@main.cpp (l_main bootstrap)");
+        //Having pushed the functions for this statement now push its parameters:
         if(lua_gettop(L) == 2)
             lua_pushnil(L);
         else
-            lua_pushvalue(L, 1);
+            lua_pushvalue(L, 1); //The executable's path.
+
+        	if(bRunDBGpClient)
```

Comment:
A couple things about this.

a.  The way this is written that code is not in the else block, but it's indented as if it were.
b.  I think this block could be pulled outside of the if(!bGotScriptFile) block so that it would be pushed regardless of whether the interpreter was specified on the command line?  I'm not 100% what's happening with the lua_State here though.